### PR TITLE
Changed SeriesGroupBy to return Series

### LINKED ISSUE #442: SeriesGroupBy.value_counts() should return `Series` not `DataFrame`
**Describe the bug**
From discussion here: https://github.com/pandas-dev/pandas-stubs/issues/203#issuecomment-1324575666

**To Reproduce**
```python
import pandas as pd

df = pd.DataFrame(
    {
        "Animal": ["Falcon", "Falcon", "Parrot", "Parrot"],
        "Max Speed": [380, 370, 24, 26],
    }
)
c: pd.Series = df.groupby("Animal")["Max Speed"].value_counts()
print(type(c))  # <class 'pandas.core.series.Series'>
print(c.index)
print(c)
```

`mypy` and `pyright` report that there is a mismatch in the result.

In `pandas` source, `SeriesGroupBy.value_counts()` always returns `Series` .   Need to make that change in typing stubs. 
https://github.com/pandas-dev/pandas-stubs/blob/44f804c7f9600f4651041a3ac002e3700bfc0890/pandas-stubs/core/groupby/generic.pyi#L75

But should copy overloads from `DataFrameGroupBy` for `value_counts()` to handle `normalize` argument correctly.

### PR DESCRIPTION
**Pull request by ramvikrams**
- [x] Closes #442 
- [x] Tests added: Please use `assert_type()` to assert the type of any return value

### CODE REVIEW

File: tests/test_frame.py [hunk 1]
```diff
@@ -1942,3 +1942,14 @@ def series_added_in_astype() -> None:
     # GH410
     df = pd.DataFrame({"a": [1, 2, 3], "b": [4, 5, 6]})
     check(assert_type(df.astype(df.dtypes), pd.DataFrame), pd.DataFrame)
+
+
+def test_SeriesGroupBy_and_value_conts() -> None:
```
Comment (gandhis1) [inline tests/test_frame.py:1947]: There is a typo here. And why is it in frame tests? There are no similar tests to put it with?

Comment (bashtage) [reply tests/test_frame.py:1947]: Also, please use standard python function naming, e.g., `test_series_groupby_and_value_counts`.

File: pandas-stubs/core/groupby/generic.pyi [hunk 2]
```diff
@@ -65,14 +65,24 @@ class SeriesGroupBy(GroupBy, Generic[S1]):
...
+    ) -> Series: ...
+    @overload
+    def value_counts(
+        self,
+        normalize: Literal[True] = ...,
+        sort: bool = ...,
+        ascending: bool = ...,
+        bins=...,
+        dropna: bool = ...,
+    ) -> Series: ...
```
[the remark this answers is not in the document]
Comment (bashtage) [reply pandas-stubs/core/groupby/generic.pyi:85]: Should not need two overloads when both return `Series`.

File: tests/test_frame.py (hunk 1 above)
Comment (ramvikrams) [reply tests/test_frame.py:1947]: yes sir

File: pandas-stubs/core/groupby/generic.pyi (hunk 2 above)
[the remark this answers is not in the document]
Comment (ramvikrams) [reply pandas-stubs/core/groupby/generic.pyi:85]: I updated the file thanks

**Comment by ramvikrams** [comment]
> thanks @ramvikrams
> 
> IN the future, please name your branches related to what you are fixing

ok sir will surely do that

**Comment by MarcoGorelli** [comment]
> But should copy overloads from DataFrameGroupBy for value_counts() to handle normalize argument correctly.

sorry @Dr-Irv , I'm confused - shouldn't overloads have been added to return `Series[float]` if `normalize=True` and `Series[int]` otherwise?

File: pandas-stubs/core/groupby/generic.pyi (hunk 2 above)
[the remark this answers is not in the document]
Comment (Dr-Irv) [reply pandas-stubs/core/groupby/generic.pyi:75]: Actually there should be two overloads here, and I realized this after merging.  @ramvikrams can you open a new PR with this:
```python
    @overload
    def value_counts(
        self,
        normalize: Literal[False] = ...,
        sort: bool = ...,
        ascending: bool = ...,
        bins=...,
        dropna: bool = ...,
    ) -> Series[int]: ...
    @overload
    def value_counts(
        self,
        normalize: Literal[True]
        sort: bool = ...,
        ascending: bool = ...,
        bins=...,
        dropna: bool = ...,
    ) -> Series[float]: ...
```

The changes from what you initially had are not doing `= ...` for `normalize` in the second overload, and adding the subtype as the return type of the two overloads.

In the test, don't do `c: pd.Series =`, and modify the test to check that it returns `Series[int]`, and then add a test with `normalize=True` and check thtat it returns `Series[float]`

**Comment by Dr-Irv** [comment]
> > But should copy overloads from DataFrameGroupBy for value_counts() to handle normalize argument correctly.
> 
> sorry @Dr-Irv , I'm confused - shouldn't overloads have been added to return `Series[float]` if `normalize=True` and `Series[int]` otherwise?

Yes, and I think I was typing the same message while you were typing yours!

File: pandas-stubs/core/groupby/generic.pyi (hunk 2 above)
[the remark this answers is not in the document]
Comment (ramvikrams) [reply pandas-stubs/core/groupby/generic.pyi:75]: Yes i'll do

[the remark this answers is not in the document]
Comment (ramvikrams) [reply pandas-stubs/core/groupby/generic.pyi:75]: > 

yes sir will do that

**Pull request merged**